addons/forum-time-zones/timezones.js

```diff
@@ -0,0 +1,144 @@
+export default async function ({ addon, console }) {
+  const twelveHourTime = addon.settings.get("twelve-hour-time") === true;
+  if (addon.settings.get("local-timezone") === true) {
+    const cache = [];
+    const locale = addon.auth.scratchLang;
+    const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+    const localCurrentTimeRepr = new Date(new Date().toLocaleString("en-US", { timeZone }));
+    const capitalize = (str) => str[0].toUpperCase() + str.slice(1);
+    const formatter = (time) => {
+      // This is the correct representation of the given time.
+      const instantTimeRepr = new Date(time);
+      // This represents the time with timezone added/subtracted to the base time.
+      // UTC representation of this is off by the timezone offset.
+      // Temporary solution until browsers implement Temporal.
+      const localTimeRepr = new Date(instantTimeRepr.toLocaleString("en-US", { timeZone }));
+      // Due to daytime saving, diff between two Date is inaccurate unless we both use timezoned Date
+      // Math.min makes sure nobody gets posts from tomorrow
+      const localDateDiff = Math.min(Math.floor(localTimeRepr / 8.64e7) - Math.floor(localCurrentTimeRepr / 8.64e7), 0);
+      const timePart = localTimeRepr.toLocaleTimeString("en-GB");
+      switch (localDateDiff) {
+        case 0:
+        case -1: {
+          if (addon.settings.get("relative-dates") === true) {
+            const relativeFormatter = new Intl.RelativeTimeFormat(locale, {
```

Comment:
```
Uncaught (in promise) TypeError: can't convert null to object
```

Reply:
For line 24?

Reply:
I got that error message on that line when going to a forum topic on Firefox.

Reply:
I'll look into it a bit later.